src/Internal/Service/CheckoutEncryptionService.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Class CheckoutEncryptionService
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Service;
+
+use WCPay\Internal\Proxy\LegacyProxy;
+
+/**
+ * Service for checkout data encryption.
+ */
+class CheckoutEncryptionService {
+
+	/**
+	 * Legacy Proxy.
+	 *
+	 * @var LegacyProxy
+	 */
+	private $legacy_proxy;
+
+	/**
+	 * Service constructor.
+	 *
+	 * @param LegacyProxy $legacy_proxy Legacy proxy.
+	 */
+	public function __construct(
+		LegacyProxy $legacy_proxy
+	) {
+		$this->legacy_proxy = $legacy_proxy;
+	}
+
+	/**
+	 * Encrypt client secret for the client.
+	 *
+	 * @param string $customer_id Customer id.
+	 * @param string $client_secret Client secret.
+	 *
+	 * @return string
+	 */
+	public function encrypt_client_secret( string $customer_id, string $client_secret ): string {
```

Comment:
If we go with the suggestion of creating a controller for managing 3DS verification URLs, maybe we could avoid having a whole service for this, and have `encrypy_client_secret` there?

Reply:
Good catch. We don't need that service. Removed it here: https://github.com/Automattic/woocommerce-payments/pull/7471/commits/8f1ebd8f2f87833dcccb8539e183ed9e164d3218